python/lsst/ts/ledprojector/led_controller.py

```diff
@@ -0,0 +1,319 @@
+# This file is part of ts_ledprojector.
+#
+# Developed for the Vera C. Rubin Observatory Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["LEDController"]
+import asyncio
+import inspect
+import logging
+import types
+
+from lsst.ts import salobj, utils
+from lsst.ts.idl.enums.run_ledprojector import (
+    LEDBasicState,
+)
+from .labjack_interface import LabJackInterface
+# Hide mypy error `Module "labjack" has no attribute "ljm"`.
+from labjack import ljm  # type: ignore
+from .mock_labjack_interface import MockLabJackInterface
+from .base_labjack_data_client import BaseLabJackDataClient
+
+# DRA-CN024D05 SSR requires 3.0VDC minimum turn-on voltage, up to 12 VDC max
+# 1.0VDC minimum turn-off voltage
+# all ports should be digital IO so write 1 for on/0 for off
+# Values to write to labjack for operating LED
+class LEDSetting():
```

Comment:
Maybe make this an enum class instead? e.g.:

```
class LEDSetting(enum.IntEnum):
    LED_OFF_SETTING = 0
    LED_ON_SETTING = 1
```

also, consider moving these ancillary things int a separate submodule.